src/Account.sol

```diff
@@ -441,287 +404,332 @@ contract Account is IAccount, OpsReady, Owned, Initializable {
+        }
+
+        // cancel gelato task
+        IOps(OPS).cancelTask({taskId: conditionalOrder.gelatoTaskId});
 
-        // /// @notice intent is targetPrice or worse despite direction
-        // if (order.sizeDelta > 0) {
-        //     // Long
-        //     return (price >= order.targetPrice, price);
-        // } else if (order.sizeDelta < 0) {
-        //     // Short
-        //     return (price <= order.targetPrice, price);
-        // }
+        // delete order from conditional orders
+        delete conditionalOrders[_conditionalOrderId];
 
-        // // sizeDelta == 0
-        // // @notice execution should never reach here
-        // // @dev needed to satisfy types
-        return (false, price);
+        events.emitConditionalOrderCancelled({
+            account: address(this),
+            conditionalOrderId: _conditionalOrderId
+        });
     }
 
     /// @inheritdoc IAccount
-    function placeOrder(
-        bytes32 _marketKey,
-        int256 _marginDelta,
-        int256 _sizeDelta,
-        uint256 _targetPrice,
-        OrderTypes _orderType,
-        uint128 _priceImpactDelta
-    ) external payable override returns (uint256) {
-        return
-            _placeOrder({
-                _marketKey: _marketKey,
-                _marginDelta: _marginDelta,
-                _sizeDelta: _sizeDelta,
-                _targetPrice: _targetPrice,
-                _orderType: _orderType,
-                _priceImpactDelta: _priceImpactDelta,
-                _maxDynamicFee: 0
-            });
+    function executeConditionalOrder(uint256 _conditionalOrderId) external override onlyOps {
+        (bool isValidConditionalOrder, uint256 fillPrice) =
+            validConditionalOrder(_conditionalOrderId);
+
+        if (!isValidConditionalOrder) {
+            revert ConditionalOrderInvalid();
+        }
+
+        ConditionalOrder memory conditionalOrder = getConditionalOrder(_conditionalOrderId);
+        address market = address(getPerpsV2Market(conditionalOrder.marketKey));
+
+        // if conditional order is reduce only, ensure position size is only reduced
+        if (conditionalOrder.reduceOnly) {
+            int256 currentSize = getPerpsV2Market(conditionalOrder.marketKey).positions({
+                account: address(this)
+            }).size;
+
+            // ensure position exists and incoming size delta is NOT the same sign
+            /// @dev if incoming size delta is the same sign, then the conditional order is not reduce only
+            if (currentSize == 0 || _isSameSign(currentSize, conditionalOrder.sizeDelta)) {
+                // remove task from gelato's side
+                /// @dev optimization done for gelato
+                IOps(OPS).cancelTask(conditionalOrder.gelatoTaskId);
+
+                // delete conditional order from conditional orders
+                delete conditionalOrders[_conditionalOrderId];
+
+                events.emitConditionalOrderCancelled({
```

Comment:
Might be good to add a reason field so we can differentiate between a user cancelling and execution cancelling.